From 30b4c55b10cdb299f2c6709adfa1ad070e788e88 Mon Sep 17 00:00:00 2001
From: Waldemar Rymarkiewicz <waldemarx.rymarkiewicz@intel.com>
Date: Tue, 13 Nov 2018 12:46:20 +0100
Subject: [PATCH] PMCPESYS-104: regulator: tps65273: Fix integer to pointer
 conversion

It has to return a pointer and not an integer. Fix compilation error:

drivers/regulator/tps65273-regulator.c:167:10: warning: return makes pointer
from integer without a cast [-Wint-conversion]   return -ENOMEM;

Signed-off-by: Waldemar Rymarkiewicz <waldemarx.rymarkiewicz@intel.com>
---
 drivers/regulator/tps65273-regulator.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/regulator/tps65273-regulator.c b/drivers/regulator/tps65273-regulator.c
index c787bfee22dd..c267d5b47355 100755
--- a/drivers/regulator/tps65273-regulator.c
+++ b/drivers/regulator/tps65273-regulator.c
@@ -163,7 +163,7 @@ static struct tps65273_platform_data *tps65273_parse_dt(struct device *dev)
 
 	rmatch = devm_kzalloc(dev, sizeof(*rmatch) * rnum, GFP_KERNEL);
 	if (!rmatch)
-		return -ENOMEM;
+		return ERR_PTR(-ENOMEM);
 
 	for (i = 0; i < rnum; i++)
 		rmatch[i].name = tps65273_regulators[i].name;
